tests/core/test_model.py

```diff
@@ -1045,6 +1046,54 @@ def test_star_expansion(assert_exp_eq) -> None:
     )
 
 
+def test_case_sensitivity(assert_exp_eq):
+    config = Config(model_defaults=ModelDefaultsConfig(dialect="snowflake"))
```

Comment:
Important note: if the config's default dialect is different from what's specified in the model, there might be an issue when doing lookup on the model's name. For example, if we had:

```sql
MODEL (
    name example.source,
    kind EMBEDDED,
    dialect snowflake,
);

SELECT
    "id",
    "name",
    "payload"
FROM db.schema."table"
```

And the config had no default dialect, doing `sqlmesh render example.source` would likely fail because the model's name is normalized into `EXAMPLE.SOURCE` and so we'll be unable to find `example.source`.

This is an issue that will probably only come up in Snowflake codebases, so maybe it's a reasonable tradeoff. The reason we can't use the model's dialect and rely on the config's default one, is that we _need_ to know the corresponding `Model` object in `Context.render` to refer to its `dialect` property, so it becomes a chicken-and-egg problem.